src/storage/mod.rs

```diff
@@ -479,6 +480,10 @@ impl<E: Engine, L: LockManager, P: PdClient + 'static> Storage<E, L, P> {
                 {
                     let begin_instant = Instant::now_coarse();
 
+                    if sample_step > 0 {
+                        limit *= sample_step;
```

Comment:
Why need we change the limit? It seems to me that the `limit` used in `Scanner::scan` limits the final result size.

Reply:
Done

Reply:
Removed.